src/actions/assetsActions.js

```diff
@@ -608,218 +595,34 @@ export const fetchAllAccountsAssetsBalancesAction = () => {
-      payload: supportedAssets,
-    });
-    dispatch(saveDbAction('supportedAssets', { supportedAssets }, true));
-  };
-};
-
-// TODO: handle side chain balances
-export const checkForMissedAssetsAction = () => {
-  return async (dispatch: Dispatch, getState: GetState) => {
-    const {
-      accounts: { data: accounts },
-      assets: { data: accountsAssets },
-    } = getState();
-
-    await dispatch(loadSupportedAssetsAction());
-    const supportedAssets = supportedAssetsSelector(getState());
-
-    const accountUpdatedAssets = accounts
-      .filter(isNotKeyBasedType)
-      .map((acc) => getSupportedTokens(supportedAssets, accountsAssets, acc))
-      .reduce((memo, { id, ...rest }) => ({ ...memo, [id]: rest }), {});
-
-    // check tx history if some assets are not enabled
-    const ownedAssetsByAccounts = await Promise.all(
-      accounts
-        .filter(isNotKeyBasedType)
-        .map(async (account) => {
-          const accountAddress = getAccountAddress(account);
-          const accountId = getAccountId(account);
-          // TODO: refactor whole checkForMissedAssetsAction
-          // $FlowFixMe: didn't refactor existing code, flow is messing up due obvious reasons
-          const ownedAssets = await etherspotService.getOwnedAssets(CHAIN.ETHEREUM, accountAddress, supportedAssets);
-          return { id: accountId, ...ownedAssets };
-        }),
-    );
-
-    const allAccountAssets = ownedAssetsByAccounts
-      .reduce((memo, { id, ...rest }) => ({ ...memo, [id]: rest }), {});
-
-    const updatedAssets = Object.keys(accountUpdatedAssets)
-      .map((acc) => ({ id: acc, ...accountUpdatedAssets[acc], ...allAccountAssets[acc] }))
-      .reduce((memo, { id, ...rest }) => ({ ...memo, [id]: rest }), {});
-
-    let newAssetsFound = false;
-    Object.keys(updatedAssets).forEach(account => {
-      if (!accountsAssets[account] || !updatedAssets[account]) {
-        newAssetsFound = true;
-        return;
-      }
-
-      const assetsInAccountBefore = Object.keys(accountsAssets[account]).length;
-      const assetsInAccountAfter = Object.keys(updatedAssets[account]).length;
+    await Promise.all(Object.keys(CHAIN).map(async (chainKey) => {
+      const chain = CHAIN[chainKey];
+      const chainSupportedAssets = await etherspotService.getChainSupportedAssets(chain);
+      // nothing to do if returned empty
+      if (isEmpty(chainSupportedAssets)) return;
 
-      if (assetsInAccountBefore !== assetsInAccountAfter) {
-        newAssetsFound = true;
-      }
-    });
-
-    if (newAssetsFound) {
       dispatch({
-        type: UPDATE_ASSETS,
-        payload: updatedAssets,
+        type: SET_CHAIN_SUPPORTED_ASSETS,
+        payload: { chain, assets: chainSupportedAssets },
       });
-      dispatch(fetchAssetsBalancesAction());
-      dispatch(saveDbAction('assets', { assets: updatedAssets }, true));
-    }
+    }));
+
+    const updatedSupportedAssets = supportedAssetsSelector(getState());
```

Comment:
Pls use direct state access here, as selector frequently do some other 

```suggestion
    const updatedSupportedAssets = getState().assets.supportedAssets;
```

Reply:
I found that using selector it's easier to replace/refactor later and there are no other pros/cons in between these 2.